### ISSUE TITLE
Annotating English words within Hebrew texts

### ISSUE THREAD

**Issue by shirawigi** [IAHLT/UD_Hebrew#47]
Hi @amir-zeldes,

We found that there are several approaches towards the annotation of foreign (mainly words that are written in English) words among us, and it seems like we need a clear guideline here.
One approach is to tag all English written words as X and assigning it only the feature Foreign.
The second approach is to allow other UPOS to such words, while also assigning the feature Foreign, along with other features such as Gender, Number etc.

examples:
A. פניצילין (**Penicillin**) היא קבוצת תרופות אנטיביוטיות מסוג בטא לקטם.

B. נכללו בו גרסה עברית לשיר "**Do You Want to Know a Secret**" של הביטלס שבמילותיו העבריות של יורם טהרלב נקרא "מזל", גרסה עברית של לאה נאור ל-"**The House of the Rising Sun**" בשם "רחוק רחוק מכאן", גרסה מחודשת לשירם של הדודאים "ערב של שושנים" וגרסה עברית לשיר "עץ הלימון".

C. בתוך קבוצה זו נמצאה גם ליידי באני (The Lady Bunny), **דראג קווין** ומי שהפכה לימים לאחת ממייסדות הפסטיבל ולמנחה שלו.

In these examples it seems very reasonable to annotate the English words "real" UPOS, deprel and feats. It seems that assigning it a UPOS X is not very informative. 
Especially when considering ex. C, in which דראג קווין is written in Hebrew but not really a Hebrew word, and it is not clear whether it's a loan word or code switching written in Hebrew, but that could easily be written in English.

If we do choose to annotate English texts with "real" UPOS, is it possible to allow it to carry the Foreign feature? Now it seems that there's a UD rule that bans all UPOS from having Foreign, except for X. 

Thanks.

**Comment by amir-zeldes** [comment]
I think it's either-or: if something is an "integrated" loan (and perhaps "drag queen" is), then it can get POS and is a loanword, not a foreign word. If it is foreign, then the guideline is to tag as X. I would say for a name like "Penicillin" we can still treat it as PROPN, since we will often have these as argument structure participants without the Hebrew equivalent. But for syntagms exhibiting internal foreign language syntax, such as "do you want to know a secret", I would tag all words in that as X and add Foreign.

**Comment by shirawigi** [comment]
@amir-zeldes  
Thanks!

About "Penicillin", didn't we agree in [issue 15](https://github.com/IAHLT/UD_Hebrew/issues/15) that it should be a simple NOUN rather than PROPN?
In which case - is the English\Latin form still should be treated as NOUN or X?

Also - if we do assign "real" UPOS to foreign texts, we think we should add the Foreign feature. Currently, the UD validator blocks us from assigning the foreign feature. Is it possible to add a rule that allows it according to UD rules?

**Comment by amir-zeldes** [comment]
Oh yes, that's right. But still, NOUN or PROPN doesn't matter for this question.

But it looks like I was premature about saying only X can be foreign, there are actually two loopholes described here:

1. If we are tagging by foreign language guidelines, i.e. tagging an English sequence, we can use Foreign=Yes and Lang=en, which would allow Foreign on that tag: https://universaldependencies.org/misc.html#lang
2. If we want to specify that something is an integrated loan but also specify its source language, we can use Foreign=Yes and OrigLang=en, see here: https://universaldependencies.org/misc.html#origlang

So basically the error would go away if we document which language this comes from, mainly because then the validator can use that language's validation rules for the validation.

**Comment by yifatbm** [comment]
@amir-zeldes, @shirawigi
Thank you!

To make this a clear guideline, can we agree that:
- words/phrases that are transcribed into HE are considered loanword (e.g. _**דראג קווין**_); --> annotate with a 'real' POS + OrigLang=en.
- words written in non-HE letters, and that have non-HE syntax (e.g. **_Do You Want to Know a Secret_**), are considered foreign; --> annotate with POS=X + Foreign=Yes
- words written in non-HE letters, but do have a HE syntax (e.g. **_רשת ה-BBC_**) are considered as loanwords; --> annotate with 'real' POS + Lang=en.

Thanks,
Shira & Yifat

**Comment by amir-zeldes** [comment]
I agree on the first two, but I'm not sure about the third:

  * words written in non-HE letters, but do have a HE syntax (e.g. רשת ה-BBC) are considered as loanwords; --> annotate with 'real' POS + Lang=en.

I think if something is just the name of that network, then it is effectively called "BBC" even in Hebrew. I would annotate PROPN and not specify the language. I'm pretty sure other UD corpora are like this too. With acronyms people don't even always know it - did you know that "ISO" is not an English acronym, but meant to be the Greek word 'iso'?

https://en.wikipedia.org/wiki/International_Organization_for_Standardization#Name_and_abbreviations

**Comment by strasss** [comment]
Everything discussed here sounds very reasonable except for the ending of Yifat's first point - 
Do were really need to tag something like דראג קווין with "OrigLang=En"? I mean, is it really relevant? 
And what about סטטוס קוו? would סטטוס also be OrigLang=Latin? And then when it's סטטוס בפייסבוק, would it be OrigLang=En? 
Seems like too much room for error somehow...

**Comment by shirawigi** [comment]
@amir-zeldes 
Thanks!

Does it mean that we never specify the source language for proper names, for example:
1. תאי T
2. תרופות מקבוצת חוסמי ה -SGLT2

Also, what should we do with cases like:
1. Penicillin V / Penicillin G
2. Penicillium glaucum / Penicillium chrysogenum

We understood that materials and diseases names should be NOUNs (not PROPNs), so probably flat(Penicillin, V) is not the correct deprel here, right? So maybe it should be npmod:desc for ex. 1 and compound for ex. 2? 

Thanks!

**Comment by Hilla-Merhav** [comment]
@amir-zeldes 
In addition to Latin loanwords as סטטוס קוו mentioned by @strasss, I am afraid the slope would become even slipperier – considering many Aramaic loanwords as דהיינו I think we need a narrower definition or stricter rules for tokens that are permitted with Foreign=Yes.

**Comment by IsraelLand** [comment]
I'm all for a clear-cut definition of where should we assign language codes and Foreign,
but on top of that, is Foreign=Yes really 1 to 1 related to X POS?
It seems UD guidelines aspire to use X very sparingly,
`The tag X is used for words that for some reason cannot be assigned a real part-of-speech category. It should be used very restrictively.
A special usage of X is for cases of code-switching where it is not possible (or meaningful) to analyze the intervening language grammatically`

I've encountered cases that still have syntactic "meaning" in Hebrew but their really impromptu borrowings are not integrated in any real way, so, like the above - cases of code-switching, but unlike the above, **can be** assigned clear POS .

**Comment by IsraelLand** [comment]
A bunch of relevant sentences are -

`האתר **פייק ריפורטר** עובד בעברית ובערבית ואוסף מידע מהציבור`
I'd assume Fake Reporter to be PROPN? Thus abiding by the matrix syntax and not the inner-embedded English syntax, otherwise we'd choose X.

`בעקבות מגפת הקורונה, מכון דוידסון פרסם מספר כללים לזיהוי **פייק ניוז** רפואי`
fake news is a noun modified by medical, adj., it seems almost impossible to treat it as X, not as a noun. Again we go for the matrix syntax.

`השקריות גם הופצה בתפוצה רחבה יותר וקיבלה **retweets** 
מיותר משתמשים יחידים בכל עומק של השתלשלות.`
In this case the inner structure is definitely English (plural form) and foreign, but cases of
`קיבל retweets  מרובים`
are absolutely possible, so again I find it hard not to treat it as a noun - even though it probably should get the Foreign=yes feature.

It's a matter of whether we go for the matrix syntax or the embedded foreign one. 

The way I see it, we tend to choose the Hebrew matrix syntax.
In Hebrew song names, we _do go_ for the embedded syntax, so
`הלחין את "הוא הלך בשדות"`  is verb, pron, adp, noun etc.,
But if the embedded syntax is meaningless, we "default" to the matrix one -
So the song name
`הלחין את Do You Want to Know a Secret`
would be PROPNs all the way, the same way we treat Fake Reporter as PROPN. The 'et' article also points to that direction imo, even though the inner syntax is meaningless, the matrix one is that of a noun/PROPN.

in cases like, 
`התוכנית נקראת אוריינות תקשורתית ("media literacy").`
Media literacy is really a phrase of its own, so a true case for X.

TL;DR - I see cases where words or phrases are foreign, but still aren't X; this seems to be in line with the idea that X [should be used sparingly](https://universaldependencies.org/u/pos/X.html).


PS - I really don't think this is a matter of the script used. Code switching is mainly a feature of spoken language, only then written down, I don't think this should make any difference.

**Comment by amir-zeldes** [comment]
OK, so there are rather many issues here and I feel like this is a good candidate for discussion at our next meeting. Generally as a TL;DR until we meet, I would like to say:

  * Things like `OrigLang` are by no means a widely used standard, and I don't necessarily recommend adding them to IAHLT data. My main intention was to say how this would be done if this were a goal - for projects I've worked onlike UD Coptic this made a lot of sense to do, since contact between Coptic, Greek and other languages is a major research objective of users of the data. But for IAHLT goals, it seems likely we should not be spending time on this; I'm happy to discuss more though
  * For things that are names, such as פייק רפורטר, I feel pretty strongly that they should be tagged PROPN, since this has implications for downstream applications, evaluation of NER and other stuff. It seems less interesting to know that internally in English פייק is somehow an `amod`, and doing a whole code-switched subtree for such cases
  * I agree X should be used sparingly, but unless we are pulling out the whole arsenal or UD multilingual treebanking (which I think we shouldn't), then X is definitely the expected handling of intervening sequences in a foreign language, such as:
    * קנדי אמר ich/X bin/X ein/X Berliner/X


OK, that became a bit long for a TL;DR :)
